pkg/controller/daemon/daemoncontroller.go

```diff
@@ -510,6 +527,10 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 	if err != nil {
 		return fmt.Errorf("couldn't get node to daemon pod mapping for daemon set %q: %v", ds.Name, err)
 	}
+	// Check DeletionTimestamp again because it may have been refreshed by getNodesToDaemonPods().
+	if ds.DeletionTimestamp != nil {
```

Comment:
See my comment in https://github.com/kubernetes/kubernetes/pull/42938/files#r105728217 for what I mean by having a top-level catch-all check.